etc/icinga2/scripts/mail-service-notification.sh

```diff
@@ -170,6 +170,9 @@ if [ "$VERBOSE" = "true" ] ; then
   logger "$PROG sends $SUBJECT => $USEREMAIL"
 fi
 
+# A mail header to relate mails for the same host/service, to enable the use of threaded view in a client.
+REFHEADER="References: <$HOSTNAME+$SERVICENAME@$(hostname --fqdn)>"
```

Comment:
First, in the other script you have used `$ICINGA2HOST` - being `hostname` - as the domain part, here you are using `$(hostname --fqdn)`. Please note that the `--fqdn` flag is not part of POSIX and thereby not available on all operating systems.

As another point, grouping all emails for the same host or service is maybe not something everyone wants. By grouping all emails for one service, reoccurring events may get lost, depending on the email client. A more useful grouping would be incident-based, as [Icinga Notifcation's Email Channel](https://github.com/Icinga/icinga-notifications) does.